identity-service/test/processNotifications/trendingTrackNotification.js

```diff
@@ -0,0 +1,179 @@
+const moment = require('moment')
+const assert = require('assert')
+const models = require('../../src/models')
+const {
+  processTrendingTracks,
+  getTimeGenreActionType,
+  TRENDING_TIME,
+  TRENDING_GENRE
+} = require('../../src/notifications/trendingTrackProcessing')
+const {
+  notificationTypes
+} = require('../../src/notifications/constants')
+
+const { clearDatabase, runMigrations } = require('../lib/app')
+
+/**
+ * Track id 100 owned by user id 1 is #1 trending
+ * Track id 101 owned by user id 1 is #2 trending
+ * Track id 102 owned by user id 2 is #3 trending
+ * Track id 103 owned by user id 3 is #4 trending
+ * Track id 104 owned by user id 4 is #5 trending
+ */
+const initialNotifications = [
+  {
+    'trackId': 100,
+    'userId': 1,
+    'rank': 1,
+    'type': notificationTypes.TrendingTrack
+  }, {
+    'trackId': 101,
+    'userId': 1,
+    'rank': 2,
+    'type': notificationTypes.TrendingTrack
+  }, {
+    'trackId': 102,
+    'userId': 2,
+    'rank': 3,
+    'type': notificationTypes.TrendingTrack
+  }, {
+    'trackId': 103,
+    'userId': 3,
+    'rank': 4,
+    'type': notificationTypes.TrendingTrack
+  }, {
+    'trackId': 104,
+    'userId': 4,
+    'rank': 5,
+    'type': notificationTypes.TrendingTrack
+  }
+]
+
+/**
+ * Track id 103 owned by user id 3 is #1 trending <= increase 
+ * Track id 104 owned by user id 4 is #2 trending <= increase
+ * Track id 100 owned by user id 1 is #3 trending <= decrease
+ * Track id 110 owned by user id 10 is #4 trending <= new
+ * Track id 101 owned by user id 1 is #5 trending <= decrease 
```

Comment:
if we exit trending and the same track re-enters to rank 3 after the interval, are we supposed to get a notif?

Reply:
No notification if the track is at the same rank, only if it's new or moved up in rank. 
Sorry it's a little confusing w/ the comment when I wrote increase I meant moved up in rank ie. from position 3 => position 1